lib/fog/azurerm/models/storage/files.rb

```diff
@@ -0,0 +1,31 @@
+require 'fog/core/collection'
+require 'fog/azurerm/models/storage/file'
+
+module Fog
+  module Storage
+    class AzureRM
+      # This class is giving implementation of listing blobs.
+      class Files < Fog::Collection
+        model Fog::Storage::AzureRM::File
+        attribute :directory
+
+        def all(options = { metadata: true })
+          files = []
+          service.list_blobs(directory, options).each do |blob|
+            hash = File.parse blob
+            hash['directory'] = directory
+            files << hash
+          end
+          load files
+        end
+
+        def get(directory, name)
+          file = File.new(service: service)
```

Comment:
True. But is there any concern about the performance? Every time we fetch a object will trigger a request. How about just leaving it to the user to call `get_properties` on demand?